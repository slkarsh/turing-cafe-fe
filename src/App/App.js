import React, { Component } from 'react';
import './App.css';
import { getReservations, postRes } from '../apiCalls/apiCalls'
import ResContainer from '../ResContainer/ResContainer'
import ResForm from '../ResForm/ResForm'

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: []
    }
  }

  componentDidMount() {
    getReservations()
      .then(res => this.setState({ reservations: res}))
  }

  addReservation = newRes => {
    // console.log('new res', newRes)
    postRes(newRes)
      .then(reservation => this.setState({ reservations: [...this.state.reservations, reservation]}))
  }

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <ResForm addReservation={this.addReservation} />
        </div>
        <div className='resy-container'>
        <ResContainer reservations={this.state.reservations} />
        </div>
       
      </div>
    )
  }
}

export default App;
