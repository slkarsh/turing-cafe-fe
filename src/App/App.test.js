import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'

describe ('App', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot();
  })
})
