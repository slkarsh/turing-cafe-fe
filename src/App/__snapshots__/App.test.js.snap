// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ResForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <ResContainer
      reservations={Array []}
    />
  </div>
</div>
`;
